### ISSUE TITLE
Prefer composable approach

### ISSUE THREAD

**Issue by roberth** [Gabriel439/haskell-nix#53]
The examples use methods that cause issues when trying to extend the package set further. Using the right tools, which are no harder, we can avoid frustration down the line.

 * Use overlays instead of overrides. They are the way to go, because the are more powerful and, I'd argue, more consistent: overlays for packages sets, overrides for packages.
 * [Avoid](https://blog.flyingcircus.io/2017/11/07/nixos-the-dos-and-donts-of-nixpkgs-overlays/) `rec` keyword. People get used to it and will use it in overlay definitions, leading to surprises.
 * Use `haskellPackages.extend` instead of `.override { overrides =`. It lets you add more layers of overrides which is more flexible. Attempting the same on the current code *replaces* the single layer of overrides with the new one, which is unexpected. As a bonus, it is syntactically simpler.

**Comment by Gabriel439** [comment]
Yeah, I agree with using overlays.  I've been using them both at work and also in some personal projects and I'm very happy with them.

I also agree with avoiding `rec` (in favor of chaining multiple overlays)

I didn't know about `extend`, though.  Let me check that out and learn how it works

**Comment by sboosali** [comment]
iirc, extend doesn't play nice with override.

**Comment by sboosali** [comment]
https://github.com/NixOS/nixpkgs/issues/26561

**Comment by sorki** [comment]
Adding two examples of Haskell overlays which you might find handy:

- https://github.com/sorki/hnix-overlay (quite simple, overlays default `haskellPackages`)
- https://github.com/HaskellEmbedded/ivory-tower-nix/ (creates a new package set called `myHaskellPackages` using specified `compiler` argument, there's also `nix/env.nix` which disables profiling for the whole set)

I'm still not sure if this is their final form as there's some clutter I would like to refactor. Feedback welcome - I will try to create a PR for this repository with the example when I'm sure about the approach.
